charts/gateway-helm/templates/namespaced-infra-manager-rbac.yaml

```diff
@@ -0,0 +1,66 @@
+{{ $watchedNamespaces := list }}
+{{ if .Values.config.envoyGateway.provider.kubernetes }}
+{{ $kube := .Values.config.envoyGateway.provider.kubernetes }}
+{{ if and ($kube.watch) ($kube.deploy) (eq $kube.deploy.type "GatewayNamespace") }}
```

Comment:
Is `$kube.watch` mandatory? 

By default, EG should watch Gateways for all the namesapces.

Reply:
I have added two RBAC modes to the helm chart:
- one is the default, when we have the GatewayNamespace Mode, it will create clusterrole for managing infra
- other is when we have selected just few namespaces to watch, it will setup only roles in the selected namespaces for managing infra